blaze-server/src/main/scala/org/http4s/blaze/server/Http1ServerStage.scala

```diff
@@ -303,17 +302,21 @@ private[blaze] class Http1ServerStage[F[_]](
             logger.trace("Request/route requested closing connection.")
             F.delay(closeConnection())
           } else
-            F.delay {
-              bodyCleanup().onComplete {
-                case s @ Success(_) => // Serve another request
-                  parser.reset()
-                  handleReqRead(s)
-
-                case Failure(EOF) => closeConnection()
-
-                case Failure(t) => fatalError(t, "Failure in body cleanup")
-              }(trampoline)
-            }
+            F.async_[Unit] { cb =>
```

Comment:
As far as I get things, it's better to use `async` when dealing with execution on an external pool (there it's a trampoline EC). Shifting back to `CE`'s EC should win.